### ISSUE TITLE
deadsnakes failing in linux build

### ISSUE THREAD

**Issue by parrt** [antlr/antlr4#1671]
Howdy. Build is now failing on linux, @antlr/antlr-targets E.g., https://travis-ci.org/antlr/antlr4/jobs/202698729

This looks to be no longer valid

```
sudo add-apt-repository ppa:fkrull/deadsnakes -y
```

but is [alive](https://launchpad.net/~fkrull/+archive/ubuntu/deadsnakes)

I see [here](https://github.com/travis-ci/apt-source-whitelist/blob/master/ubuntu.json):

```
 {
    "alias": "deadsnakes",
    "sourceline": "ppa:fkrull/deadsnakes",
    "key_url": null
  },
```

that null is suspicious. Can whoever added this figure this out? thanks!

**Comment by bhamiltoncx** [comment]
It appears @ericvergnaud added this in https://github.com/antlr/antlr4/commit/4f48d533fa0e557856eb66267525397365037e97 .

**Comment by parrt** [comment]
Ah. Ok, yeah, seems like travis has somehow disabled this. not sure what the workaround is. Eric will awaken in a few more hours in HK :)

**Comment by ericvergnaud** [comment]
Hi,

this was actually added long before, to enable install of specific versions of Python.

This was working with build #1205 a few hours ago, and is most likely a travis bug.
I would request support from them rather than look for a workaround.

that said, the below line:
  sudo add-apt-repository ppa:fkrull/deadsnakes -y
should only be required for the Python3 target 

similarly the below line:
  sudo add-apt-repository ppa:rwky/nodejs -y
should only be required for the Node target 

I don’t have time now to try removing these lines in other targets (and watch the travis builds), but feel free to do so if it helps get back to an acceptable situation.

Eric

**Comment by parrt** [comment]
Dang. travis looks broken still. I see this in docs:

```
The following snippet shows these three types of APT sources

addons:
  apt:
    sources:
    - deadsnakes
    - sourceline: 'ppa:ubuntu-toolchain-r/test'
    - sourceline: 'deb https://packagecloud.io/chef/stable/ubuntu/precise main'
      key_url: 'https://packagecloud.io/gpg.key'
```

Then, I see:

```
addons:
  apt:
    sources:
    - ubuntu-toolchain-r-test
    packages:
    - gcc-4.8
    - g++-4.8
```

**Comment by ericvergnaud** [comment]
PR #1676 reduces the size of the problem
looks like the issue originates from rely/nodejs, not deadsnakes

**Comment by ericvergnaud** [comment]
Fixed with latest commit in PR #1676
